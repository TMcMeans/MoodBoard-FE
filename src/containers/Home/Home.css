/*COMPONENT STYLES*/
.home {
  display: flex;
  flex-direction: column;
  align-items: center;
}

/*TITLE*/
.welcome {
  color: #ffffff;
  margin: 0;
}

.this-month {
  color: #ffffff;
}

/*ANIMATION*/
.bubbles {
  background-color: white;
  height: 40vh;
  margin-bottom: 2em;
  width: 95%;
}

/*BUTTONS*/
.home-btns-container {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin-top: 1rem;
  width: 80%;
}

.home-btns {
  height: 3rem;
  line-height: 1rem;
  width: 150px;
}
