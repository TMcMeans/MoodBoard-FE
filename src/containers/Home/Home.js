import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import Logo from '../../components/Logo/Logo';
import { getTonesByMonth } from '../../thunks/getTonesByMonth';
import './Home.css';

class Home extends Component {
  constructor() {
    super();
    this.state = {
      date: new Date()
    };
  }

  formatDate = () => {
    const { date } = this.state;
    let year = date.getFullYear();
    let month = date.getMonth() + 1;

    if (month < 10) {
      month = '0' + month.toString();
    }

    return year.toString() + '-' + month;
  }

  componentDidMount = async () => {
    const urlDate = this.formatDate();
    const url = `https://mood-board-be.herokuapp.com/api/v1/users/1/tone_responses?date=${urlDate}`;
    await this.props.getTonesByMonth(url);
    // set the tones to colors
    
    // assign colors to individual days (by the day's value)
  }

  render() {
    const name = 'Ash';
    return (
      <div className="home">
        <Logo />
        <h1 className="welcome">Welcome, {name}!</h1>
        <h3 className="this-month">This month...</h3>
        <div className="bubbles" />
        <div className="home-btns-container">
          <Link to="/journal">
            <button className="standard-btn light home-btns">journal</button>
          </Link>
          <Link to="/calendarView">
            <button className="standard-btn light home-btns">calendar</button>
          </Link>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  tones: state.tones
})

const mapDispatchToProps = dispatch => ({
  getTonesByMonth: (url) => dispatch(getTonesByMonth(url))
})

export default connect(mapStateToProps, mapDispatchToProps)(Home);
