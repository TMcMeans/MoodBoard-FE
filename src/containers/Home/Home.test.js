import React from 'react';
import { shallow } from 'enzyme';
import Home from './Home';

describe('Home', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Home />);
  });
  it('should render a snapshot with correct data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
