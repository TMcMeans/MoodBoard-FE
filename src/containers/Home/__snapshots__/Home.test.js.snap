// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render a snapshot with correct data passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Logo />,
        <h1
          className="welcome"
        >
          Welcome, 
          Ash
          !
        </h1>,
        <h3
          className="this-month"
        >
          This month...
        </h3>,
        <div
          className="bubbles"
        />,
        <Link
          replace={false}
          to="/journal"
        >
          <button>
            journal
          </button>
        </Link>,
        <Link
          replace={false}
          to="/calendarView"
        >
          <button>
            calendar
          </button>
        </Link>,
      ],
      "className": "home",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Welcome, ",
            "Ash",
            "!",
          ],
          "className": "welcome",
        },
        "ref": null,
        "rendered": Array [
          "Welcome, ",
          "Ash",
          "!",
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "This month...",
          "className": "this-month",
        },
        "ref": null,
        "rendered": "This month...",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "bubbles",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button>
            journal
          </button>,
          "replace": false,
          "to": "/journal",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "journal",
          },
          "ref": null,
          "rendered": "journal",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button>
            calendar
          </button>,
          "replace": false,
          "to": "/calendarView",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "calendar",
          },
          "ref": null,
          "rendered": "calendar",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Logo />,
          <h1
            className="welcome"
          >
            Welcome, 
            Ash
            !
          </h1>,
          <h3
            className="this-month"
          >
            This month...
          </h3>,
          <div
            className="bubbles"
          />,
          <Link
            replace={false}
            to="/journal"
          >
            <button>
              journal
            </button>
          </Link>,
          <Link
            replace={false}
            to="/calendarView"
          >
            <button>
              calendar
            </button>
          </Link>,
        ],
        "className": "home",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Welcome, ",
              "Ash",
              "!",
            ],
            "className": "welcome",
          },
          "ref": null,
          "rendered": Array [
            "Welcome, ",
            "Ash",
            "!",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This month...",
            "className": "this-month",
          },
          "ref": null,
          "rendered": "This month...",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "bubbles",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button>
              journal
            </button>,
            "replace": false,
            "to": "/journal",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "journal",
            },
            "ref": null,
            "rendered": "journal",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button>
              calendar
            </button>,
            "replace": false,
            "to": "/calendarView",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "calendar",
            },
            "ref": null,
            "rendered": "calendar",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
